russstation/code/datums/components/dwarf_rune.dm

```diff
@@ -28,14 +29,23 @@
 	rune.expend()
 	to_chat(user, span_notice("You rub \the [attacker] on [source] and it becomes enchanted."))
 
-/datum/component/dwarf_rune/proc/onItemAttack(datum/source, atom/movable/target, mob/living/user)
+/datum/component/dwarf_rune/proc/onItemAttack(datum/source, mob/living/target, mob/living/user)
 	SIGNAL_HANDLER
 
 	return checkAttack(user)
 
+/datum/component/dwarf_rune/proc/onItemAttackObj(datum/source, atom/movable/target, mob/living/user)
+	SIGNAL_HANDLER
+
+	// for some reason these aren't always the right types?? why tg why
+	if(istype(user))
+		return checkAttack(user)
+	else if(istype(target, /mob/living))
```

Comment:
```suggestion
	else if(isliving(target))
```

Reply:
also I have no idea what you mean by "aren't always the right types" but it sounds cursed

Reply:
i legit don't get it, but i was getting runtimes like `/obj/whatever doesn't have dropItemToGround()` when hitting things with the dwarf tool, which meant that the obj being attacked was in the `user` param. def cursed